source/server/sv_client.c

```diff
@@ -1264,6 +1264,14 @@ void SV_ParseClientMessage( client_t *client, msg_t *msg )
 					break;
```

Comment:
I am not sure how long `Steam_BeginAuthSession` takes, but during this command you already have `client->steamid` set to input from a client, and it is not accepted by steam yet.
I would suggest to store it in a temp variable first,
and only store it when `Steam_BeginAuthSession` accepts it.

Reply:
In the current configuration, Steam_BeginAuthSession is blocking, and an untrusted steamid will instantly be rejected by the DropClient. This is probably not ideal for the server, I should look into getting some sort of event loop set up